extensions/gmsa-dc/v1/Setup-gMSA.ps1

```diff
@@ -0,0 +1,271 @@
+﻿<#
+ .SYNOPSIS
+ This script will automate the creation of a Group Managed Service Account,
+ install the account locally, produce the JSON file with gMSA info, and convert
+ it to the Yaml file needed.
+
+ .NOTES
+ This is only for automated e2e testing.  DO NOT use this for production.
+ Jeremy Wood (JeremyWx)
+ Version: 1.0.0.0
+#>
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference = 'Stop'
+$PSDefaultParameterValues['*:ErrorAction'] = 'Stop'
+
+# Set script variables - These will need changed if this extension is moved to a new repo!!!
+$GMSARoot = "C:\gmsa"
+
+# Function to check AD services
+function CheckADServices()
+	{
+	Sleep 4
+	$svcs = "adws","dns","kdc","netlogon","ntds","lanmanserver","lanmanworkstation"
+    $svcstatus = 0
+        foreach ( $svc in $svcs ) {
+            if ( $(Get-Service -Name $svc).Status -notlike "Running") {
+                $svcstatus += 1
+            }
+        }
+
+	if ($svcstatus -gt 0) {
+		return $false
+		} else {
+		return $true
+		}
+	}
+
+# Function to make sure AD is fully up
+function CheckGroup()
+    {
+	try {
+		Sleep 2
+	    $x = Get-AdGroupMember -Identity "Enterprise Admins"
```

Comment:
nit: indentation here looks off for `$x = Get-AdGroupMember -Identity "Enterprise Admins"` . Please check the indentation.